crates/plugins/src/manager.rs

```diff
@@ -194,6 +194,48 @@ impl PluginManager {
         };
         Ok(plugin_manifest)
     }
+
+    pub async fn update_lock(&self) -> PluginManagerUpdateLock {
+        let lock = self.update_lock_impl().await;
+        PluginManagerUpdateLock { lock }
+    }
+
+    async fn update_lock_impl(&self) -> anyhow::Result<fd_lock::RwLock<tokio::fs::File>> {
+        let plugins_dir = self.store().get_plugins_directory();
+        tokio::fs::create_dir_all(plugins_dir).await?;
+        let file = tokio::fs::File::create(plugins_dir.join(".updatelock")).await?;
+        let locker = fd_lock::RwLock::new(file);
+        Ok(locker)
+    }
+}
+
+pub struct PluginManagerUpdateLock {
+    lock: anyhow::Result<fd_lock::RwLock<tokio::fs::File>>,
```

Comment:
Why do we keep a `Result` here?

Reply:
I am also not sure about this one. Looks like any errors in `update_lock_impl` would be unexpected and should be surfaced.

Reply:
It seems this is juggling 3 scenarios:

1. The lock is successfully created and acquired :tada:
2. The lock is successfully created but already held; assume that another update is already happening and bail
3. The lock cannot be created; continue without the lock

Reply:
What @lann said.  The idea is to forbid the update only if we _positively_ detect that another update is in progress - error cases are "proceed because at least it's no worse than what we have today."  We could of course discard the Results and Options and use our own enums to track the case, and that would certainly make it more self-documenting...!